src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -46,65 +45,106 @@ public RelationshipPersistenceService(
         this.conversionService = conversionService;
     }
 
-    // region Create
+    /**
+     * Create a relationship.
+     *
+     * @param tenantUrn the tenant URN
+     * @param createRelationship the relationship to create
+     * @return
+     */
     @Override
     public Optional<RelationshipResponse> create(String tenantUrn, RelationshipCreate createRelationship) {
 
-        if (alreadyExists(tenantUrn, createRelationship)) {
-            return Optional.empty();
-        }
-        RelationshipEntity entity = null;
         try {
+            // If the requested object already exists, return Optional.empty()
+            Optional<RelationshipResponse> alreadyExists = findSpecific(
+                tenantUrn,
+                createRelationship.getSource().getType(),
+                createRelationship.getSource().getUrn(),
+                createRelationship.getTarget().getType(),
+                createRelationship.getTarget().getUrn(),
+                createRelationship.getRelationshipType());
+
+            if (alreadyExists.isPresent()) {
+                return Optional.empty();
+            }
+
             UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
-            entity = conversionService.convert(createRelationship, RelationshipEntity.class);
+            RelationshipEntity entity = conversionService.convert(createRelationship, RelationshipEntity.class);
             entity.setTenantId(tenantId);
             entity = persist(entity);
+            return Optional.ofNullable(conversionService.convert(entity, RelationshipResponse.class));
 
-        } catch (Exception e) {
-            exceptionLogger("RelationshipPersistenceService:create", e, tenantUrn, createRelationship);
+        } catch (IllegalArgumentException | ConversionException e) {
+            String msg = String.format("create failed, tenant: '%s', cause: %s", tenantUrn, e.toString());
+            log.error(msg);
+            log.debug(msg, e);
             throw e;
```

Comment:
This is where I would normally throw a platform specific Exception, e.g.
```
throw new SmartCosmosException("Exception creating relationship: '" +createRelationship.toString() + "', cause: " e.toString()
```
It makes the interface for all methods more consistent, keeps the logic for the caller simple.

In fact the best implementation is an aspect but that is a topic for another day.